src/libcore/num/f32.rs

```diff
@@ -461,4 +461,124 @@ impl f32 {
         // It turns out the safety issues with sNaN were overblown! Hooray!
         unsafe { mem::transmute(v) }
     }
+
+    /// Return the memory representation of this floating point number as a byte array in
+    /// big-endian (network) byte order.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(float_to_from_bytes)]
+    /// let bytes = 3.141565f32.to_be_bytes();
+    /// assert_eq!(bytes, [0x40, 0x49, 0x0f, 0x67]);
+    /// ```
+    #[unstable(feature = "float_to_from_bytes", issue = "60446")]
+    #[inline]
+    pub fn to_be_bytes(self) -> [u8; 4] {
+        self.to_bits().to_be_bytes()
+    }
+
+    /// Return the memory representation of this floating point number as a byte array in
+    /// little-endian byte order.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(float_to_from_bytes)]
+    /// let bytes = 3.141565f32.to_le_bytes();
+    /// assert_eq!(bytes, [0x67, 0x0f, 0x49, 0x40]);
+    /// ```
+    #[unstable(feature = "float_to_from_bytes", issue = "60446")]
+    #[inline]
+    pub fn to_le_bytes(self) -> [u8; 4] {
+        self.to_bits().to_le_bytes()
+    }
+
+    /// Return the memory representation of this floating point number as a byte array in
+    /// native byte order.
+    ///
+    /// As the target platform's native endianness is used, portable code
+    /// should use [`to_be_bytes`] or [`to_le_bytes`], as appropriate, instead.
+    ///
+    /// [`to_be_bytes`]: #method.to_be_bytes
+    /// [`to_le_bytes`]: #method.to_le_bytes
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(float_to_from_bytes)]
+    /// let bytes = 3.141565f32.to_ne_bytes();
+    /// assert_eq!(
+    ///     bytes,
+    ///     if cfg!(target_endian = "big") {
+    ///         [0x40, 0x49, 0x0f, 0x67]
+    ///     } else {
+    ///         [0x67, 0x0f, 0x49, 0x40]
+    ///     }
+    /// );
+    /// ```
+    #[unstable(feature = "float_to_from_bytes", issue = "60446")]
+    #[inline]
+    pub fn to_ne_bytes(self) -> [u8; 4] {
+        self.to_bits().to_ne_bytes()
+    }
+
+    /// Create a floating point value from its representation as a byte array in big endian.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(float_to_from_bytes)]
+    /// let value = f32::from_be_bytes([0x40, 0x49, 0x0f, 0x67]);
+    /// let difference = (value - 3.141565f32).abs();
+    /// assert!(difference <= 1e-5);
```

Comment:
Why not `==`?

Reply:
I did the same check in from_bits method: https://github.com/rust-lang/rust/blob/e6c2844cad866f5495bffdb51d634a187c34ab76/src/libcore/num/f32.rs#L456-L463

Reply:
Also I think that floating point equality comparison is incorrect and
discourage.

Reply:
> Also I think that floating point equality comparison is incorrect and
> discourage.

Normally this is true, but if you just created the float using a bit pattern, it must be actually equal to the expected value.

Reply:
I will let the decision to Simon.

Reply:
Floating point equality can give unexpected results when it’s done on the result of non-trivial computation, because precision errors can acumulate. For example `0.3+0.3+0.3 != 0.9`.

But as @tbu- said, the conversions here should be exact. Please change to `from_*_bytes` and `from_bits` doctests to use `assert_eq!`, and let’s see if the tests pass.

Also, please imitate `from_bits` / `to_bits` in picking a floating point value like 12.5 that has a “nice” representation, and using hex literals for integers.

Reply:
I has a concern: What if people follow our style in their tests for 
floating point equality?

Reply:
> Normally this is true, but if you just created the float using a bit pattern, it must be actually equal to the expected value.

Do we guarantee this for NaNs ? cc @rkruppe

Reply:
I highly doubt we have anything relevant explicitly written down. I don't know whether LLVM explicitly guarantees it for bitcasts only. I don't see why it wouldn't, but who knows. But even if LLVM explicitly stated that, that wouldn't mean we'd translate that into any Rust level guarantee.

Reply:
FWIW, this PR does not introduce this problem. This is a problem that `f32::from_bits` and `f64::from_bits` already have, and these are stable, so I don't think clarifying this should hold this PR back. If we document this for those methods, that would just follow for these as well.